pandapipes/component_models/abstract_models/branch_models.py

```diff
@@ -265,33 +265,111 @@ def calculate_pressure_lift(cls, net, pipe_pit, node_pit):
         raise NotImplementedError
 
     @classmethod
-    def calculate_temperature_lift(cls, net, pipe_pit, node_pit):
+    def calculate_temperature_lift(cls, net, branch_pit, node_pit):
         """
 
         :param net:
         :type net:
-        :param pipe_pit:
-        :type pipe_pit:
+        :param branch_pit:
+        :type branch_pit:
         :param node_pit:
         :type node_pit:
         :return:
         :rtype:
         """
         raise NotImplementedError
 
+
     @classmethod
-    def extract_results(cls, net, options, node_name):
-        results = super().extract_results(net, options, node_name)
+    def prepare_result_tables(cls, net, options, node_name):
+        res_table = super().extract_results(net, options, node_name)
 
         f, t = get_lookup(net, "branch", "from_to")[cls.table_name()]
         fa, ta = get_lookup(net, "branch", "from_to_active")[cls.table_name()]
 
         placement_table = np.argsort(net[cls.table_name()].index.values)
         idx_pit = net["_pit"]["branch"][f:t, ELEMENT_IDX]
         pipe_considered = get_lookup(net, "branch", "active")[f:t]
-        idx_sort, active_pipes = _sum_by_group(idx_pit, pipe_considered.astype(np.int32))
+        _, active_pipes = _sum_by_group(idx_pit, pipe_considered.astype(np.int32))
         active_pipes = active_pipes > 0.99
         placement_table = placement_table[active_pipes]
         branch_pit = net["_active_pit"]["branch"][fa:ta, :]
+        return placement_table, branch_pit, res_table
+
 
-        return placement_table, branch_pit, results
+    @classmethod
+    def extract_results(cls, net, options, node_name):
```

Comment:
Was this moved from the pipe component? Why can this be generalized?

Reply:
Yes, this move was from pipe component (and also more or less from heat_exchanger and valve, as they have the same extract results function). The main reason why I did this was the following: I realized that in pipe component there were changes in extract_results which I couldn't find in valve and heat exchange component (duplicated code problem -.-). Therefore I decided to bring it up so that all of them inherit the same function (their common class is the branch_component). I agree that it seems to be not generalizable, but I bringt up the following argument: I say that from all branch_components these extract results are the most general. If not, you need to adapt it as user.

Alternatively, in order to prevent duplicated code I was thinking about to move it in a toolbox function which all of them can just use. But I prefer this solution more as it keeps the most relevant parts all at one place.